File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.Draft.BatchPrint/MethodExtensions.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace SolidEdge.Draft.BatchPrint
{
    public static class FileInfoExtensions
    {
        public static Icon GetSmallIcon(this FileInfo fi)
        {
            int flags = NativeMethods.SHGFI_ICON | NativeMethods.SHGFI_SMALLICON;
            int fileAttributes = NativeMethods.FILE_ATTRIBUTE_NORMAL;

            NativeMethods.SHFILEINFO shfi = new NativeMethods.SHFILEINFO();
            IntPtr result = NativeMethods.SHGetFileInfo(
                fi.FullName,
                fileAttributes,
                out shfi,
                Marshal.SizeOf(shfi),
                flags);

            if (result.ToInt32() == 0)
            {
                return null;
            }
            else
            {
                return Icon.FromHandle(shfi.hIcon);
            }
        }
    }
}

File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.Draft.BatchPrint/MethodExtensions.vb
Imports System.IO
Imports System.Runtime.InteropServices
Imports System.Text

Namespace SolidEdge.Draft.BatchPrint
	Public Module FileInfoExtensions
		<System.Runtime.CompilerServices.Extension> _
		Public Function GetSmallIcon(ByVal fi As FileInfo) As Icon
			Dim flags As Integer = NativeMethods.SHGFI_ICON Or NativeMethods.SHGFI_SMALLICON
			Dim fileAttributes As Integer = NativeMethods.FILE_ATTRIBUTE_NORMAL

			Dim shfi As New NativeMethods.SHFILEINFO()
			Dim result As IntPtr = NativeMethods.SHGetFileInfo(fi.FullName, fileAttributes, shfi, Marshal.SizeOf(shfi), flags)

			If result.ToInt32() = 0 Then
				Return Nothing
			Else
				Return Icon.FromHandle(shfi.hIcon)
			End If
		End Function
	End Module
End Namespace
